pkg/cmd/update-tls-artifacts/generate-owners/tlsmetadatainterfaces/annotation_requirement.go

```diff
@@ -63,38 +64,44 @@ func (o annotationRequirement) InspectRequirement(rawData []*certgraphapi.PKILis
 		violationJSONBytes)
 }
 
-func (o annotationRequirement) generateInspectionMarkdown(pkiInfo *certgraphapi.PKIRegistryInfo) ([]byte, error) {
-	compliantCertsByOwner := map[string][]certgraphapi.PKIRegistryInClusterCertKeyPair{}
-	violatingCertsByOwner := map[string][]certgraphapi.PKIRegistryInClusterCertKeyPair{}
-	compliantCABundlesByOwner := map[string][]certgraphapi.PKIRegistryInClusterCABundle{}
-	violatingCABundlesByOwner := map[string][]certgraphapi.PKIRegistryInClusterCABundle{}
-
-	for i := range pkiInfo.CertKeyPairs {
-		curr := pkiInfo.CertKeyPairs[i]
-		owner := curr.CertKeyInfo.OwningJiraComponent
-		regenerates, _ := AnnotationValue(curr.CertKeyInfo.SelectedCertMetadataAnnotations, o.GetAnnotationName())
-		if len(regenerates) == 0 {
-			violatingCertsByOwner[owner] = append(violatingCertsByOwner[owner], curr)
-			continue
-		}
-
-		compliantCertsByOwner[owner] = append(compliantCertsByOwner[owner], curr)
-	}
-	for i := range pkiInfo.CertificateAuthorityBundles {
-		curr := pkiInfo.CertificateAuthorityBundles[i]
-		owner := curr.CABundleInfo.OwningJiraComponent
-		regenerates, _ := AnnotationValue(curr.CABundleInfo.SelectedCertMetadataAnnotations, o.GetAnnotationName())
-		if len(regenerates) == 0 {
-			violatingCABundlesByOwner[owner] = append(violatingCABundlesByOwner[owner], curr)
-			continue
-		}
-		compliantCABundlesByOwner[owner] = append(compliantCABundlesByOwner[owner], curr)
-	}
+func (o annotationRequirement) generateInspectionMarkdown(pkiInfo *certs.PKIRegistryInfo) ([]byte, error) {
+	complianceIntermediate := BuildAnnotationComplianceIntermediate(pkiInfo, InspectAnnotationHasValue(o.GetAnnotationName()))
+	compliantCertsByOwner := complianceIntermediate.CompliantCertsByOwner
+	violatingCertsByOwner := complianceIntermediate.ViolatingCertsByOwner
+	compliantCABundlesByOwner := complianceIntermediate.CompliantCABundlesByOwner
+	violatingCABundlesByOwner := complianceIntermediate.ViolatingCABundlesByOwner
 
 	md := NewMarkdown(o.title)
 	md.Title(2, "How to meet the requirement")
 	md.ExactText(o.explanationMD)
 
+	certsWithoutOwners := violatingCertsByOwner[UnknownOwner]
+	caBundlesWithoutOwners := violatingCABundlesByOwner[UnknownOwner]
+
+	if len(certsWithoutOwners) > 0 || len(caBundlesWithoutOwners) > 0 {
```

Comment:
This makes sense in ownership, but I think we want to see the main sort remain, compliant or non-compliant in places like auto-regenerate-after-expiry